tests/main/security-dev-input-event-denied/task.yaml

```diff
@@ -0,0 +1,55 @@
+summary: Ensure that /dev/input/event* is denied by default.
+
+details: |
+    The default policy disallows access to /dev/input/event*.
+
+    The joystick interface disallows access to /dev/input/event* for
+    non-joysticks.
+
+    The test checks the snap is not able to access /dev/input/event* with or
+    without the joystick interface connected. We do this since the
+    /dev/input/event* devices are sensitive and because the joystick interface
+    adds a /dev/input/event* AppArmor glob rule that relies entirely on the
+    device cgroup for enforcement.
+
+prepare: |
+    echo "Given the test-snapd-event snap is installed"
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-event
+
+restore: |
+    rm -f call.error
+
+execute: |
+    echo "The joystick plug is not connected by default"
+    snap interfaces -i joystick | MATCH '\- +test-snapd-event:joystick'
+
+    if [ "$(snap debug confinement)" != "strict" ]; then
+        exit 0
+    fi
+
+    echo "Then the snap is not able to access an evdev keyboard"
+    if test-snapd-event "-event-kbd" 2>"${PWD}"/call.error; then
+        echo "Expected permission error calling evtest with disconnected plug"
+        exit 1
+    fi
+    # AppArmor is 'Permission denied' which is expected with default policy
+    MATCH "Permission denied" < call.error
+
+    echo "When the joystick plug is connected"
+    snap connect test-snapd-event:joystick
+    udevadm settle
```

Comment:
This is actually what prompted the settle PR-- I noticed that I was able to access the device because of a race between when the apparmor profile was loaded into the kernel and udev handled the triggered events. If the other PR is committed, this settle can be removed.